### ISSUE TITLE
Attention mask not applied in action head (N1.5?)

### ISSUE THREAD

**Issue by Caesar107** [NVIDIA/Isaac-GR00T#417]
Hi, I noticed that the action head doesn’t seem to use the attention mask in cross attention. Is this intentional or a missing implementation detail?
```python
for idx, block in enumerate(self.transformer_blocks):
    if idx % 2 == 1 and self.config.interleave_self_attention:
        hidden_states = block(
            hidden_states,
            attention_mask=None,
            encoder_hidden_states=None,
            encoder_attention_mask=None,
            temb=temb,
        )
    else:
        hidden_states = block(
            hidden_states,
            attention_mask=None,
            encoder_hidden_states=encoder_hidden_states,
            encoder_attention_mask=None,
            temb=temb,
        )
    all_hidden_states.append(hidden_states)
